src/pages/iv-item-parts/forms/ItemPartForm.js

```diff
@@ -0,0 +1,177 @@
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { useForm } from 'src/hooks/form'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { Grid } from '@mui/material'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+
+const ItemPartForm = ({ labels, maxAccess, obj }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { formik } = useForm({
+    enableReinitialize: true,
+    validateOnChange: true,
+
+    validationSchema: yup.object({
+      items: yup
+        .array()
+        .of(
+          yup.object().shape({
+            qty: yup.number().test(function (value) {
+              return isRowEmpty(this.parent) || value > 0
+            })
+          })
+        )
+        .required()
+    }),
+    initialValues: {
+      itemId: obj.recordId,
+      items: [
+        {
+          id: 1,
+          partRef: '',
+          partname: '',
+          partId: '',
+          qty: ''
+        }
+      ]
+    },
+    onSubmit: values => {
+      const { items } = values
+
+      if (items.length === 1 && isRowEmpty(items[0])) {
+        postData({ itemId: values.itemId, items: [] })
+      } else {
+        postData(values)
+      }
+    }
+  })
+
+  const isRowEmpty = row => {
+    return !row.partId
+  }
+
+  const postData = async obj => {
+    const itemsToPost = obj?.items?.map((item, index) => ({
+      ...item,
+      itemId: obj.itemId,
+      seqNo: index + 1
+    }))
+
+    const data = {
+      itemId: obj.itemId,
+      items: itemsToPost || []
+    }
+
+    await postRequest({
+      extension: InventoryRepository.ItemParts.set2,
+      record: JSON.stringify(data)
+    })
+
+    toast.success(platformLabels.Updated)
+  }
+
+  const columns = [
+    {
+      component: 'resourcelookup',
+      label: labels.part,
+      name: 'partRef',
+      props: {
+        valueField: 'reference',
+        displayField: 'reference',
+        displayFieldWidth: 2,
+        endpointId: InventoryRepository.Parts.snapshot,
+        mapping: [
+          { from: 'recordId', to: 'partId' },
+          { from: 'name', to: 'partName' },
+          { from: 'reference', to: 'partRef' }
+        ],
+        columnsInDropDown: [
+          { key: 'reference', value: 'Reference' },
+          { key: 'name', value: 'Name' }
+        ]
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.partName,
+      name: 'partName',
+      props: {
+        readOnly: true,
+        allowNegative: false,
+        decimalScale: 0
+      }
+    },
+    {
+      component: 'numberfield',
+      label: labels.qty,
+      name: 'qty',
+      props: {
+        allowNegative: false,
+        decimalScale: 0
+      }
+    }
+  ]
+
+  function getData() {
```

Comment:
If you are certain that you do not need getData somewhere in your code, you can use useEffect directly and insert the method getData in useEffect.